internal/config/config.go

```diff
@@ -153,6 +135,14 @@ func (c *AuthConfig) HasEnvToken() bool {
 			return true
 		}
 	}
+	// TODO: This is _extremely_ knowledgable about the implementation of TokenFromEnvOrConfig
+	// It has to use a hostname that is not going to be found in the hosts so that it
+	// can guarantee that tokens will only be returned from a set env var.
+	// Discussed here, but maybe worth revisiting: https://github.com/cli/cli/pull/7169#discussion_r1136979033
+	//
+	// By providing example.com. it's also _only_ looking for GH_ENTERPRISE_TOKEN or GITHUB_ENTERPRISE_TOKEN
```

Comment:
It checks `GH_TOKEN` and `GITHUB_TOKEN` as well.

Reply:
Ohhh sorry, I missed that the `isEnterprise` branch of `tokenForHost` would fall through. Thanks for correcting me!